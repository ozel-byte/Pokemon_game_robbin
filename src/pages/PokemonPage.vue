<template>
  <div class="fondo">
    <h1 v-if="!pokemon">Espere por favor...</h1>
    <div v-else>
      <h1>Â¿Quien es este pokemon?</h1>
      <pokemon-img :pokemonId="pokemon.id" :showPokemon="showPokemon" />
      <pokemon-opcion :pokemons="pokemonArr" @selection="checkAnswer" />

      <template v-if="showAnser">
        <h2>{{ message }}</h2>
        <button @click="newGame">Nuevo Juego</button>
      </template>
    </div>
  </div>
</template>

<script>
import PokemonImg from "@/components/PokemonImg.vue";
import PokemonOpcion from "@/components/PokemonOpcion.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
  name: "Pokemon Page",
  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
      message: "",
      showAnser: false,
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonArr[rndInt];
    },
    newGame() {
      this.showAnser = false;
      this.showPokemon = false;
      this.pokemonArr = [];
      this.pokemon = null;
      this.mixPokemonArray();
    },
    checkAnswer(pokemonId) {
      this.showPokemon = true;
      if (pokemonId === this.pokemon.id) {
        this.showAnser = true;
        this.message = `Respuesta correcta ${this.pokemon.name}`;
      } else {
        this.showAnser = true;
        this.message = `Respuesta incorrecta la opcion era ${this.pokemon.name}`;
      }
    },
  },
  mounted() {
    this.mixPokemonArray();
  },
  components: {
    PokemonImg,
    PokemonOpcion,
  },
};
</script>

<style scope>
#app {
  padding: 0px;
  margin: 0px;
}
html,
body {
  text-align: center;
}
h1{
  padding: 10px;
  margin: 0px;
  color: #862938;
  font-family: Arial, Helvetica, sans-serif;
}
.fondo {
  margin: 0px;
  padding: 0px;
  height: 100vh;
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
button{
  background-color: #f6e7e2;
  color: #832634;
  border: none;
  padding: 10px;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bold;
  cursor: pointer;
}
</style>