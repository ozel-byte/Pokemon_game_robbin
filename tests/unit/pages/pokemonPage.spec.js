import {shallowMount,mount} from '@vue/test-utils';
import PokemonPage from '@/pages/PokemonPage';


describe( 'pokemonPage', () => {
    let wrapper;


    beforeEach( () => {
        wrapper=shallowMount(PokemonPage);
    });

    test( 'debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot();
    });
    test( 'debe de llamar mixPokemonArr al montar', () => {
        const mixpokemonarray = jest.spyOn( PokemonPage.methods, 'mixPokemonArray');
        const wrapper = shallowMount(PokemonPage);
        expect(mixpokemonarray).toHaveBeenCalled();
    });
    test( 'debe de hacer match con el snapshot cuando cargan los pokemons', () => {
        const wrapper = shallowMount(PokemonPage,{
            data(){
                return {
                    pokemonArr: [
                        {name: 'bulbasaur', id: 1},
                        {name: 'ivysaur', id:   2},
                        {name: 'venusaur', id:  3},
                        {name: 'charmander', id: 4}
                    ],
                    pokemon: null,
                    showPokemon: false,
                    message: "",
                    showAnser: false,   
                }
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('debe de mostrar los componentes de PokemonPicture y PokemonOptions', () => {

        //PokemonPicture debe de existir
        const wrapper =  shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr: [
                        {name: 'bulbasaur', id: 1},
                        {name: 'ivysaur', id:   2},
                        {name: 'venusaur', id:  3},
                        {name: 'charmander', id: 5}
                    ],
                    showPokemon: false,
                    message: "",
                    pokemon: {
                        name: 'charmander',
                        id: 5
                    },
                    showAnser: false, 
                }
            }
        });
        console.log(wrapper.html());
        const t = wrapper.findComponent("pokemon-img-stub");
        expect(t.exists()).toBeTruthy();
        //PokemonOptions debe de existir

        expect(wrapper.findComponent("pokemon-opcion-stub").exists()).toBeTruthy();
        //PokemonPicture atrribute pokemoinid = 5
        console.log(t.attributes('pokemonid'));
        expect(t.attributes('pokemonid')).toBe('5')
        //PokemonOptions tenga atrribute pokemons tobe true
        console.log(wrapper.findComponent('pokemon-opcion-stub').attributes());
        expect(wrapper.findComponent('pokemon-opcion-stub').attributes('pokemons')).toBeTruthy()

    });

    test( ' Pruebas con checkAnswer', async () => {
        const wrapper =  shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr: [
                        {name: 'bulbasaur', id: 1},
                        {name: 'ivysaur', id:   2},
                        {name: 'venusaur', id:  3},
                        {name: 'charmander', id: 5}
                    ],
                    showPokemon: false,
                    message: "",
                    pokemon: {
                        name: 'charmander',
                        id: 5
                    },
                    showAnser: false, 
                }
            }
        });

       await wrapper.vm.checkAnswer(5);

       expect( wrapper.find('h2').exists()).toBeTruthy();
       expect(wrapper.vm.showPokemon).toBe(true);
       expect(wrapper.find('h2').text()).toBe(`Respuesta correcta charmander`)
    });
});