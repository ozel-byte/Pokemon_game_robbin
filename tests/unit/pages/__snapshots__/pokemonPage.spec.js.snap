// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonPage debe de hacer match con el snapshot 1`] = `
<div class="fondo">
  <h1>Espere por favor...</h1>
</div>
`;

exports[`pokemonPage debe de hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div class="fondo">
  <h1>Espere por favor...</h1>
</div>
`;
